src/shader.cpp

```diff
@@ -689,14 +689,35 @@ ShaderInfo generate_shader(std::string name, u8 material_type, u8 drawtype,
 		float sample_step;
 		int smooth = (int)g_settings->getFloat("normalmaps_smooth");
 		switch (smooth){
-		case 0:
-			sample_step = 0.0078125; // 1.0 / 128.0
-			break;
 		case 1:
-			sample_step = 0.00390625; // 1.0 / 256.0
+			sample_step = 0.5; // 1/2
 			break;
 		case 2:
-			sample_step = 0.001953125; // 1.0 / 512.0
+			sample_step = 0.25; // 1/4
+			break;
+		case 3:
+			sample_step = 0.125; // 1/8
+			break;
+		case 4:
+			sample_step = 0.0625; // 1/16
+			break;
+		case 5:
+			sample_step = 0.03125; // 1/32
+			break;
+		case 6:
+			sample_step = 0.015625; // 1/64
+			break;
+		case 7:
+			sample_step = 0.0078125; // 1/128
+			break;
+		case 8:
+			sample_step = 0.00390625; // 1/256
+			break;
+		case 9:
+			sample_step = 0.001953125; // 1/512
+			break;
+		case 10:
+			sample_step = 0.0009765625; // 1/1024
```

Comment:
This should use something more general like `sample_step = 1 / pow(2, smooth);` (with some sanity checks of course).